src/main/java/com/terracottatech/frs/RestartStoreFactory.java

```diff
@@ -52,46 +52,54 @@ public abstract class RestartStoreFactory {
   private RestartStoreFactory() {
   }
 
-  private static ActionCodec<ByteBuffer, ByteBuffer, ByteBuffer> createCodec(ObjectManager<ByteBuffer, ByteBuffer, ByteBuffer> objectManager) {
-    ActionCodec<ByteBuffer, ByteBuffer, ByteBuffer> codec =
-            new ActionCodecImpl<ByteBuffer, ByteBuffer, ByteBuffer>(objectManager);
+  private static ActionCodec<ByteBuffer, ByteBuffer, ByteBuffer> createCodec(
+      ObjectManager<ByteBuffer, ByteBuffer, ByteBuffer> objectManager,
+      CipherManager cipherManager) {
+    ActionCodec<ByteBuffer, ByteBuffer, ByteBuffer> codec = new ActionCodecImpl<>(objectManager, cipherManager);
```

Comment:
If you introduce an `EncryptionActions.registerActions(...)` then you pass the `CipherManager` to it, and not have to use the ActionCodec as a holder.